code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -1027,6 +1024,7 @@
 /obj/vehicle/sealed/mecha/mob_exit(mob/M, silent, forced)
 	var/newloc = get_turf(src)
 	var/atom/movable/mob_container
+	update_icon()
```

Comment:
This doesnt  fix anthing move it to later in the proc and remove the other updateicons